lib/discovery.js

```diff
@@ -406,18 +408,12 @@ function mixinDiscovery(MySQL, mysql) {
 
   MySQL.prototype.buildPropertyType = function(columnDefinition) {
     var mysqlType = columnDefinition.dataType;
+    var columnType = columnDefinition.columnType;
     var dataLength = columnDefinition.dataLength;
 
     var type = mysqlType.toUpperCase();
     switch (type) {
       case 'CHAR':
-        if (dataLength === 1) {
-          // Treat char(1) as boolean
-          return 'Boolean';
-        } else {
-          return 'String';
-        }
-        break;
```

Comment:
Backwards compatibility is important, please preserve it.

> I wonder if it would be possible to add a switch to change the behavior?

I am fine with that, this is the approach we are usually taking - add a configuration option allowing users to opt into backwards-incompatible behaviour.